backend/inspirehep/migrator/tasks.py

```diff
@@ -275,30 +276,41 @@ def create_records_from_mirror_recids(self, recids):
     """
     models_committed.disconnect(index_after_commit)
     processed_records = set()
-    for recid in recids:
-        try:
-            LOGGER.info("Migrate record from mirror", recid=recid)
-            with db.session.begin_nested():
-                record = migrate_record_from_mirror(
-                    LegacyRecordsMirror.query.get(recid)
-                )
-            if record:
-                processed_records.add(str(record.id))
-            else:
-                LOGGER.warning("Record is empty", recid=recid)
-        except OperationalError:
-            LOGGER.exception("Got operational error", recid=recid)
-            raise
-        except Exception:
-            LOGGER.exception("Cannot migrate record", recid=recid)
-            continue
     try:
+        for recid in recids:
+            try:
+                LOGGER.info("Migrate record from mirror", recid=recid)
+                with db.session.begin_nested():
+                    record = migrate_record_from_mirror(
+                        LegacyRecordsMirror.query.get(recid)
+                    )
+                if record:
+                    processed_records.add(str(record.id))
+                else:
+                    LOGGER.warning("Record is empty", recid=recid)
+            except OperationalError:
+                LOGGER.exception("Got operational error", recid=recid)
+                raise
+            except Exception:
+                LOGGER.exception("Cannot migrate record", recid=recid)
+                raise
+
         db.session.commit()
+    except (InvalidRequestError, OperationalError, StatementError):
```

Comment:
:+1: you also need to update https://github.com/inspirehep/inspirehep/pull/792/files#diff-2718fbe727bdf849c6819bf9d3ffa4d6R268

Reply:
:+1: